src/components/ValidateCode/ExpiredValidateCodeModal.js

```diff
@@ -0,0 +1,112 @@
+import React, {PureComponent} from 'react';
+import {withOnyx} from 'react-native-onyx';
+import _, {compose} from 'underscore';
+import lodashGet from 'lodash/get';
+import {View} from 'react-native';
+import colors from '../../styles/colors';
+import styles from '../../styles/styles';
+import Icon from '../Icon';
+import withLocalize, {withLocalizePropTypes} from '../withLocalize';
+import Text from '../Text';
+import * as Expensicons from '../Icon/Expensicons';
+import * as Illustrations from '../Icon/Illustrations';
+import variables from '../../styles/variables';
+import TextLink from '../TextLink';
+import ONYXKEYS from '../../ONYXKEYS';
+import * as ErrorUtils from '../../libs/ErrorUtils';
+import * as Session from '../../libs/actions/Session';
+
+const propTypes = {
+    ...withLocalizePropTypes,
+};
+
+class ExpiredValidateCodeModal extends PureComponent {
+    constructor(props) {
+        super(props);
+
+        this.requestNewCode = this.requestNewCode.bind(this);
+    }
+
+    shouldShowRequestCodeLink() {
+        return Boolean(lodashGet(this.props, 'credentials.login', null));
+    }
+
+    requestNewCode() {
+        Session.resendValidateCode();
+    }
+
+    render() {
+        const codeRequestedMessage = lodashGet(this.props, 'account.message', null);
+        const accountErrors = lodashGet(this.props, 'account.errors', {});
```

Comment:
NAB but I think we can simplify this part as `getLatestErrorMessage` already checks for keys and if it's empty it will return an empty string which should evaluate to false. no?

```suggestion
        const codeRequestedErrors = ErrorUtils.getLatestErrorMessage(this.props.account);
```